src/lstring.c

```diff
@@ -46,6 +46,29 @@ lstr_t *str_init(const char *str)
     }
     pstr->size = lsize(str);
     pstr->buffer = (char *)malloc(pstr->size + 1);
-    lstr_cpy(pstr->buffer, str, 3, 4, 3);
+    lstr_cpy(pstr->buffer, str, 3, 4, 1);
     return pstr;
 }
+
+// Function to concatenate ch2 string to ch1 string
+
+char *Lstrcat(char *ch1, const char *ch2)
```

Comment:
I’m not sure if it’s the best approach, but you might consider using the `realloc` function for resizing the new lstr_t.